code/game/objects/items/weapons/mop.dm

```diff
@@ -12,6 +12,13 @@
 	matter = list(MATERIAL_PLASTIC = 3)
 	var/mopping = 0
 	var/mopcount = 0
+	//check is too bulky, so here a list. Just add stuff here
+	var/get_wet_containers = list(/obj/item/weapon/reagent_containers/glass/bucket,
```

Comment:
This whitelist is a bad approach.

I mentioned previously the best method to do it, there's a proc. atom.is_open_container. It's defined in game/atoms.dm
It checks a flag set on an object, but it can also be overridden for custom behaviour
![opencontainer](https://i.imgur.com/UXG8its.png)

Have the mop call that proc on the target atom, and take some reagents from it if it returns true. Remove the whitelist and instead set the correct flag on anything that doesn't have it